File: Java/Q66-Plus-One.java
class Solution {
    public int[] plusOne(int[] digits) {
        int[] arr=null;
        int c=0;
        for(int i=digits.length-1;i>=0;i--){
            if(digits[i]==9){
                digits[i]=0;
            }else {
                digits[i]++;
                return digits;
            }
        }
        arr=new int[digits.length+1];
        arr[0]=1;
        return arr;
        
    }
}

File: Typescript/Q66-Plus-One.ts
function plusOne(digits: number[]): number[] {
  for (let i = digits.length - 1; i >= 0; i--) {
    if (digits[i] == 9) {
      digits[i] = 0;
    } else {
      digits[i]++;
      return digits;
    }
  }
  var ar = Array(digits.length + 1).fill(0);
  ar[0] = 1;
  return ar;
}
